content/en/docs/ops/deployment/multicluster-considerations/index.md

```diff
@@ -0,0 +1,155 @@
+
+{{< image width="80%"
+    link="configuration.svg"
+    caption="Primary clusters with independent sources of configuration"
+    >}}
+
+This model has the following advantages:
+
+* Reduces the blast radius of bad configuration and helps to maintain high availability for the control plane.
+* Allows administrators more fine-grained control over configuration rollout (e.g. one cluster at a time). For large
+meshes with many clusters, this process can be automated with CI/CD pipelines.
+* Allows selective exposure of `Service`/`ServiceEntry` across cluster boundaries. For example, an administrator may
+choose to deploy the HelloWorld service to Cluster A, but not Cluster B. This is an advanced use case, but may be
+desirable for stronger isolation between clusters.
+
+This duplication of configuration across primary clusters can lead to issues, however. In particular, Kubernetes
+Services must be deployed to each cluster that {{< gloss "service consumer">}}consumes{{</ gloss >}} the service
+in order to allow DNS lookup to succeed. See [DNS and Service Deployment](#dns-and-service-deployment) for more
+information.
+
+### Endpoint Discovery
+
+An Istio control plane manages traffic within the mesh by providing each proxy with the list of service endpoints.
+In order to make this work in a multicluster scenario, each control plane must observe endpoints from the API Server
+in every cluster.
+
+To enable endpoint discovery across the mesh:
+
+* Create a remote secret for each cluster, which grants access to its API server.
+* Deploy each remote secret to each primary cluster (within the Istio system namespace).
+
+The control planes will then connect and discover the service endpoints from each cluster, enabling cross-cluster
+load balancing for these services.
+
+{{< image width="80%"
+    link="endpoint-discovery.svg"
+    caption="Primary clusters with endpoint discovery"
+    >}}
+
+By default, Istio will load balance requests evenly between endpoints in each cluster. In large systems that span
+geographic regions, it may be desirable to use locality load balancing to prefer that traffic stay in the same zone
+or region.
+
+In some advanced scenarios, load balancing across clusters may not be desired. For example, in a blue/green
+deployment, you may deploy different versions of the system to different clusters. In this case, each cluster is
+effectively operating as an independent mesh. This behavior can be achieved in a couple of ways:
+
+* Do not exchange remote secrets between the clusters. This offers the strongest isolation between the clusters.
+* Use `VirtualService` and `DestinationRule` to disallow routing between two versions of the services.
+
+In either case, cross-cluster load balancing is prevented. External traffic can be routed to one cluster or the
+other using an external load balancer.
+
+{{< image width="80%"
+    link="blue-green.svg"
+    caption="Blue-green deployment without cross-cluster load balancing"
+    >}}
+
```

Comment:
Regarding blue green deployment, is this still 1 single mesh? I could argue this is 2 meshes.

My preference is to remove endpoint discovery section.  It is great detail and I love it, I just think it is not a consideration for multicluster istio.  endpoint discovery is required else it is not 1 mesh.

Reply:
Funny thing ... I *think* that I added this section based on your feedback in the gdoc :). You're right, of course... it is effectively a separate mesh and I say so above. I do believe this still belongs here, however, because blue-green deployment is often how multicluster upgrades are performed.

Reply:
My bad if I was asking different things. :-(  Maybe I think i was asking how to fold the replicated control plane piece in while we want to maintain the boundary of the clusters and allow admins to selectively expose and consume services acorss clusters.  

Looking at your preview url, I didn't get you are saying this is separate mesh (it is marked as 1 mesh in the diagram).  If there is a case this is useful for multicluster upgrade, i think we should highlight it here.  It is not obvious to me atm when reading it.  I don't understand how this relates to multicluster from users' perspective as to when to choose not to enable endpoint discovery.  Are you saying you recommend undo "create-remote-secrete' on the to-be-upgraded cluster?

Reply:
ah good point WRT the diagram. Thing is, we *still* don't have a really solid definition of when it's a single mesh vs multi-mesh. In the case of blue-green, they very well could be using the same root of trust ... in which case, *maybe* they're still conceptually a single mesh?

@smawson WDYT?